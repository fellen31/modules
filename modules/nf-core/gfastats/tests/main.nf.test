nextflow_process {

    name "Test Process GFASTATS"
    script "../main.nf"
    process "GFASTATS"

    tag "modules"
    tag "modules_nfcore"
    tag "gfastats"

    test("test_gfastats_fasta_gfa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'gfa'   // GFA output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = []      // No exclude bed file
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_gfa_fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fasta' // Fasta output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = []      // No exclude bed file
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_include_bed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fastq' // Fastq output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                input[6] = []      // No exclude bed file
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_exclude_bed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fasta' // Fasta output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_sak") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fasta' // Fasta output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = []      // No exclude bed file
                input[7] = Channel.of(
                    'RVCP\tMT192765.1'
                    ).collectFile(name:'assembly.sak')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_fasta_gfa - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'gfa'   // GFA output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = []      // No exclude bed file
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_gfa_fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fasta' // Fasta output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = []      // No exclude bed file
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_include_bed - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fastq' // Fastq output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                input[6] = []      // No exclude bed file
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_exclude_bed - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fasta' // Fasta output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                input[7] = []       // No swiss army knife instructions
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_gfastats_sak - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'fasta' // Fasta output format
                input[2] = ''      // No genome size
                input[3] = ''      // No target
                input[4] = []      // No agp file
                input[5] = []      // No include bed file
                input[6] = []      // No exclude bed file
                input[7] = Channel.of(
                    'RVCP\tMT192765.1'
                    ).collectFile(name:'assembly.sak')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
