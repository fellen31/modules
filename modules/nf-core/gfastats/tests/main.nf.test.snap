{
    "test_gfastats_gfa_fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:36:56.592934142"
    },
    "test_gfastats_fasta_gfa": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gfa.gz:md5,7cbd79c3c7ec561d11f71104e0914401"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gfa.gz:md5,7cbd79c3c7ec561d11f71104e0914401"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:16:09.02196294"
    },
    "test_gfastats_exclude_bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:25:49.419061372"
    },
    "test_gfastats_fasta_gfa - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gfa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gfa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:36:46.62046346"
    },
    "test_gfastats_sak - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:37:26.038089265"
    },
    "test_gfastats_include_bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:37:05.969933937"
    },
    "test_gfastats_exclude_bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:37:15.758306588"
    },
    "test_gfastats_include_bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:25:34.679010127"
    },
    "test_gfastats_gfa_fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:25:24.209327792"
    },
    "test_gfastats_sak": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "assembly_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly_summary:md5,3dcd4e83d6d6d5fca917ea56eb546ec3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2c78a9131c5aeb50341d0c8890a4f4d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:26:01.032032115"
    }
}